samza-core/src/main/java/org/apache/samza/runtime/LocalApplicationRunner.java

```diff
@@ -59,13 +67,23 @@
 public class LocalApplicationRunner implements ApplicationRunner {
 
   private static final Logger LOG = LoggerFactory.getLogger(LocalApplicationRunner.class);
+  private static final String RUNID_PATH = "runId";
+  private static final String APPLICATION_RUNNER_PATH_SUFFIX = "/ApplicationRunnerData";
+  private static final String RUNID_LOCK_ID = "runId";
+  private static final int LOCK_TIMEOUT = 10;
+  private static final TimeUnit LOCK_TIMEOUT_UNIT = TimeUnit.MINUTES;
 
   private final ApplicationDescriptorImpl<? extends ApplicationDescriptor> appDesc;
   private final LocalJobPlanner planner;
   private final Set<StreamProcessor> processors = ConcurrentHashMap.newKeySet();
   private final CountDownLatch shutdownLatch = new CountDownLatch(1);
   private final AtomicInteger numProcessorsToStart = new AtomicInteger();
   private final AtomicReference<Throwable> failure = new AtomicReference<>();
+  private final String uid = UUID.randomUUID().toString();
+  private CoordinationUtils coordinationUtils = null;
+  private DistributedReadWriteLock runIdLock = null;
+  private DistributedDataAccess runIdAccess = null;
+  private String runId = null;
```

Comment:
make final. Assign null if needed in the constructor.

Reply:
same issue as above. runIdLock and runIdAccess are not inited in case of STREAM and not in the constructor. runId is also inited in a separate method and not the constructor.